.github/workflows/release.yml

```diff
@@ -0,0 +1,42 @@
+name: Semantic Release
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read # for checkout
+
+jobs:
+  release:
+    name: Semantic Release
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write # to be able to publish a GitHub release
+      issues: write # to be able to comment on released issues
+      pull-requests: write # to be able to comment on released pull requests
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        uses: pnpm/action-setup@v4
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'pnpm'
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build app
+        run: pnpm build
```

Comment:
I don't think those are needed.

Reply:
I want to make sure tests pass, and that it builds. I was even going to add the docker build to be absolutely sure that it pass too.

Reply:
The tests shall pass **before** things end up on main.

we don't have any GHA credits, double check with Nath before running time-heavy workflows. /cc @iDmple

Reply:
That workflow should only bump the version(s) and publish a tag, imho.

Reply:
Ok. There is a client hook running tests & build before any commit, so yeah, it is probably safe.